wg-charter.html

```diff
@@ -0,0 +1,406 @@
+            efficiently processing graphics and related features.
+        </p>
+
+        <p>
+            This Working Group will recommend a Web programming interface for
+            graphics and computation that:
+        </p>
+
+        <ol>
+            <li>enables rendering of modern graphics to both onscreen and offscreen drawing surfaces</li>
+            <li>enables computation tasks to be performed, and the results of such tasks to be retrieved</li>
+            <li>This group may also recommend a companion Shading Language that describes the graphics and computation tasks in a format
+            that can be translated or compiled into platform-specific instructions</li>
+        </ol>
+
+        <p>
+            The API will not be restricted to any particular platform technology. Instead, it will
+            be generic enough to be implemented on top of modern GPU system APIs, such as
+            Microsoft's Direct3D 12, Apple's Metal, and Khronos's Vulkan.
+        </p>
+
+        <p>
+            Note that the majority of the input to this Working Group will come
+            directly from the <a href="https://www.w3.org/community/gpu/">GPU for the Web Community Group</a>.
+            No major development will happen in the Working Group itself. Instead, the Community
+            Group will be driving the technical work.
+        </p>
+
+        <h2>Out of Scope</h2>
+        <p>
+            The scope of work is restricted to the development of a programming
+            interface between the Web Platform and modern 3D graphics and
+            computation capabilities present on native system platforms. The work
+            will not define hardware features or algorithms, and the interface it
+            defines is not intended to be directly exposed by a GPU driver.
+        </p>
+
+      </section>
+
+
+
+      <section id="deliverables">
+        <h2>
+          Deliverables
+        </h2>
+
+        <div id="normative">
+          <h3>
+            Normative Specifications
+          </h3>
+          <dl>
+          <dt>WebGPU API</dt>
+
+          <dd>
+              An API for performing operations, such as rendering and
+              computation, on a Graphics Processing Unit (GPU).
+              <p class="draft-status"><b>Draft state:</b> <a href="https://gpuweb.github.io/gpuweb/">Adopted from the GPU for the Web Community Group</a></p>
+              <p class="milestone"><b>Recommendation expected completion:</b> Q3 2021</p>
+          </dd>
+          </dl>
+
+          <h3>
+            Optional Specifications
+          </h3>
+          <dl>
+          <dt>WebGPU Shading Language</dt>
+
+          <dd>
+              A Shading Language specification that defines the programmable
+              interface to the GPU for the Web graphics and computation
+              pipeline. Alternatively, the Working Group may recommend
+              an existing Shading Language.
```

Comment:
It seems that since starting work on this charter, the CG has firmed up their stance on the Shader Language. May want to consider updating this prose to reflect the latest consensus from the CG. (Be more precise, instead of offering the "Alternatively..." option.)

Reply:
Note the PR I prepared firms things up, moving the spec to the list of deliverables... or is too firm now?

Reply:
My comment was mostly about the sentence here starting with "Alternatively". Is the WG still going to recommend an existing Shading Language? (I don't think so.)